### PR TITLE
fix: Refactor Robotoff methods

### PR DESCRIPTION
**Pull request by atharv028**
### What
- Deprecated and routed the robotoff apis to a new file for decluttering
- Changed names of some APIs to maintain consistency in new file
- No breaking changes for now

### Screenshot
Screenshot 2023-04-05 at 12 40 39 AM

### Fixes bug(s)
- #714 

### Part of 
- #339

### CODE REVIEW

File: lib/src/robotoff_api_client.dart [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    );
+
+    return InsightsResult.fromJson(
+      HttpHelper().jsonDecode(utf8.decode(response.bodyBytes)),
+    );
+  }
+
+  static Future<RobotoffQuestionResult> getProductQuestions(
+    String barcode,
+    String lang, {
```
Comment (monsieurtanuki) [inline lib/src/robotoff_api_client.dart:71]: Should be `OpenFoodFactsLanguage language` instead.

File: lib/src/robotoff_api_client.dart [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    );
+    var result = RobotoffQuestionResult.fromJson(
+      HttpHelper().jsonDecode(utf8.decode(response.bodyBytes)),
+    );
+
+    return result;
+  }
+
+  static Future<RobotoffQuestionResult> getRandomQuestions(
+    OpenFoodFactsLanguage? lang,
```
Comment (monsieurtanuki) [inline lib/src/robotoff_api_client.dart:104]: Should be `language`.

**Review by monsieurtanuki — COMMENTED** [review]
Thank you @atharv028!
Please have a look at my comments.

File: lib/openfoodfacts.dart [hunk 3]
```diff
@@ -107,4 +107,5 @@ export 'src/utils/tag_type.dart';
 export 'src/utils/unit_helper.dart';
 export 'src/utils/uri_helper.dart';
 export 'src/utils/uri_reader.dart';
+export 'src/robotoff_api_client.dart';
```
[the remark this answers is not in the document]
Comment (atharv028) [reply lib/openfoodfacts.dart]: Done

File: lib/src/open_food_api_client.dart [hunk 4]
```diff
@@ -788,226 +789,66 @@ class OpenFoodAPIClient {
     }
   }
 
-  //TODO(x): Add comments for robotoff
+  //TODO : Remove Robotoff APIs From here
```
[the remark this answers is not in the document]
Comment (atharv028) [reply lib/src/open_food_api_client.dart]: Added TODO on each method in `open_food_api_client.dart` which routes to RobotOffAPIClient

File: lib/src/open_food_api_client.dart [hunk 5]
```diff
@@ -2,6 +2,7 @@ import 'dart:async';
 import 'dart:convert';
 
 import 'package:http/http.dart';
+import 'package:openfoodfacts/openfoodfacts.dart';
```
[the remark this answers is not in the document]
Comment (atharv028) [reply lib/src/open_food_api_client.dart]: Fixed

File: lib/src/robotoff_api_client.dart [hunk 6]
```diff
@@ -0,0 +1,182 @@
+import 'dart:convert';
+
+import 'package:http/http.dart';
+import 'package:openfoodfacts/openfoodfacts.dart';
+
+class RobotoffApiClient {
```
[the remark this answers is not in the document]
Comment (atharv028) [reply lib/src/robotoff_api_client.dart]: Changed the class name to `RobotOffAPIClient `

File: lib/src/robotoff_api_client.dart [hunk 7]
```diff
@@ -0,0 +1,182 @@
+import 'dart:convert';
+
+import 'package:http/http.dart';
+import 'package:openfoodfacts/openfoodfacts.dart';
```
[the remark this answers is not in the document]
Comment (atharv028) [reply lib/src/robotoff_api_client.dart]: Fixed

File: lib/src/robotoff_api_client.dart [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+
+class RobotoffApiClient {
+  RobotoffApiClient._();
+
+  //TODO(x): Add comments for robotoff APIs
```
[the remark this answers is not in the document]
Comment (atharv028) [reply lib/src/robotoff_api_client.dart]: Deleted the comment

File: lib/src/robotoff_api_client.dart (hunk 1 above)
Comment (atharv028) [reply lib/src/robotoff_api_client.dart]: Changed the parameter type to `OpenFoodFactsLanguage ` and name to `language `

File: lib/src/robotoff_api_client.dart (hunk 2 above)
Comment (atharv028) [reply lib/src/robotoff_api_client.dart]: Done

**Review by monsieurtanuki — APPROVED** [review]
Thank you @atharv028!

**Pull request merged**